internal/config/ca_config/ca_config.go

```diff
@@ -0,0 +1,50 @@
+package ca_config
+
+type CAIdType int
+
+const (
+	InternalCA CAIdType = iota + 1
+	AsyncInternalCA
+)
+
+type InternalCACfg struct {
+	CaCertFile         string `json:"caCertFile,omitempty"`
+	CaKeyFile          string `json:"caKeyFile,omitempty"`
+	SignerCertName     string `json:"SignerCertName,omitempty"`
+	CaSerialFile       string `json:"caSerialFile,omitempty"`
+	CaCertValidityDays int    `json:"CaCertValidityDays,omitempty"`
+	CACertStore        string `json:"CACertStore,omitempty"`
+}
+
+type CAConfigType struct {
+	CAType                          CAIdType       `json:"CAType,omitempty"`
+	AdminCommonName                 string         `json:"AdminCommonName,omitempty"`
+	ClientBootstrapCommonName       string         `json:"ClientBootstrapCommonName,omitempty"`
+	ClientBootstrapCertName         string         `json:"ClientBootstrapCertName,omitempty"`
+	ClientBootstrapCommonNamePrefix string         `json:"ClientBootstrapCommonNamePrefix,omitempty"`
+	ClientBootstrapValidityDays     int            `json:"ClientBootStrapValidityDays,omitempty"`
+	DeviceCommonNamePrefix          string         `json:"DeviceCommonNamePrefix,omitempty"`
+	InternalCAConfig                *InternalCACfg `json:"InternalCAConfig,omitempty"`
+	ServerCertValidityDays          int            `json:"ServerCertValidityDays,omitempty"`
+	ExtraAllowedPrefixes            []string       `json:"ExtraAllowedPrefixes,omitempty"`
+}
+
+func NewDefault(certStore string) *CAConfigType {
+	c := &CAConfigType{
+		CAType:                          InternalCA,
+		AdminCommonName:                 "flightctl-admin",
+		ClientBootstrapCertName:         "client-enrollment",
+		ClientBootstrapCommonName:       "client-enrollment",
+		ClientBootstrapCommonNamePrefix: "client-enrollment-",
```

Comment:
Any chance to change this to `client-enrollment:` (for consistency) without breaking things?

Reply:
The checks for prefixes were rewritten to allow "add this historic values to the prefix list" - it will accept a historic prefix while using the new one for new entries. 
So - yes. Once this is in - yes. You can migrate a to a different prefix and set the old one in the AdditionalPrefixes list